packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -443,6 +518,50 @@ export default function(
     }
   }
 
+  function callGetDerivedStateFromProps(workInProgress, instance, props) {
+    const {type} = workInProgress;
+
+    if (typeof type.getDerivedStateFromProps === 'function') {
+      if (__DEV__) {
+        if (
+          typeof instance.componentWillReceiveProps === 'function' ||
+          typeof instance.UNSAFE_componentWillReceiveProps === 'function'
+        ) {
+          const componentName = getComponentName(workInProgress) || 'Unknown';
+          if (!didWarnAboutWillReceivePropsAndDerivedState[componentName]) {
+            warning(
+              false,
+              '%s: Defines both componentWillReceiveProps() and static ' +
+                'getDerivedStateFromProps() methods. We recommend using ' +
+                'only getDerivedStateFromProps().',
+              componentName,
+            );
+            didWarnAboutWillReceivePropsAndDerivedState[componentName] = true;
+          }
+        }
+      }
+
+      const partialState = type.getDerivedStateFromProps(props, instance.state);
```

Comment:
How do you feel about `.call(null` so that people can't rely on `this` value being the `type`? Not that it's hugely important, but could be easy to accidentally break in the future.

Reply:
I initially mirrored [the check in `ReactFiberUpdateQueue`](https://github.com/facebook/react/blob/bd6b533c29cc617b4ddfc0a8b623e782c468cd52/packages/react-reconciler/src/ReactFiberUpdateQueue.js#L286), but I think it's worth changing here. 👍  Will do.